torch/_torch_docs.py

```diff
@@ -5182,13 +5182,24 @@ def merge_dicts(*dicts):
     >>> torch.quantile(a, q)
     tensor([-0.5446,  0.0700,  0.9214])
 
-.. function:: quantile(input, q, dim=None, keepdim=False, *, out=None) -> Tensor
+.. function:: quantile(input, q, dim=None, interpolation='linear', keepdim=False, *, out=None) -> Tensor
 
-Returns the q-th quantiles of each row of the :attr:`input` tensor along the dimension
-:attr:`dim`, doing a linear interpolation when the q-th quantile lies between two
-data points. By default, :attr:`dim` is ``None`` resulting in the :attr:`input` tensor
+Returns the q-th quantiles of each row of the :attr:`input` tensor
+along the dimension :attr:`dim` based on :attr:`interpolation`.
+When the desired quantile lies between two data points ``i < j``,
+the result is computed based on the :attr:`interpolation` value as described below.
+By default, :attr:`interpolation` is ``linear`` and :attr:`dim` is ``None`` resulting in the :attr:`input` tensor
 being flattened before computation.
 
+When the quantile value lies between two data points ``i < j``,
+the result is computed according to the given :attr:`interpolation` method as follows:
```

Comment:
I'm a little confused by how the language works here. Unfortunately mathematical language is often unclear. I think we may even need to revise the first sentence of this doc.

In particular, what is the quantile? There are two numbers here, the specification of the cut, which is something like `.25`, and the value we associate with that cut which has the property that at most 25% of the values in reduced row are below it and at least 75% (rounded up) of the values in the reduced row are above it. Note that it's important we not claim that 25% of values are below it and 75% of values are above. For example, if we have 3 values `{0, 1, 2}` and we take the cut `.8` then the value we associate with that cut will be between `1` and `2` and has the property that 66% of values are below it and 33% are above it. I think this is the right way to think about the computation?

So, back to "what is the quantile?" Is the quantile the specification of the cut, the `.25` or the associated value, like `1.5`, which has the given property? Or is the quantile the pair of them? Or is quantile inherently confusing and the language we use here should just be more specific about about converting a mapping from `[0, 1]` to a set of values with this property?

Reply:
This is a good question. Quantile as Wikipedia defines it are cut points that divide the range of a distribution into segments of equal probability (median is a 2-quantile dividing into 2 segments of 50% probability). However, what this operator computes is a single quantile value such that no more than q elements are smaller than the computed quantile and no more than (1 - q) are bigger than the computed quantile.

Reply:
To answer your question more directly, the quantile is the value that cuts the data. So the 0.5 quantile of an input [1, 2, 3] is 2.

Reply:
OK, so we're finding the quantiles (values) that satisfy the property stipulated by the input. That makes sense.